tests/geometry/costs_test.py

```diff
@@ -138,12 +138,44 @@ class TestTICost:
           costs.SqEuclidean()
       ]
   )
-  def test_h_transform(self, rng: jax.Array, cost_fn: costs.TICost):
-    x = jax.random.normal(rng, (15, 3))
-    h_transform = cost_fn.h_transform(mu.logsumexp)
-    h_transform = jax.jit(jax.vmap(jax.grad(h_transform)))
+  def test_transport_map(self, rng: jax.Array, cost_fn: costs.TICost):
+    d = 5
+    rng_x, rng_A = jax.random.split(rng)
+    x = jax.random.normal(rng_x, (15, d))
+    A = jax.random.normal(rng_A, (d, d * 2))
+    A = A @ A.T
 
-    np.testing.assert_array_equal(jnp.isfinite(h_transform(x)), True)
+    transport_fn = cost_fn.transport_map(lambda z: -jnp.sum(z * (A.dot(z))))
+    transport_fn = jax.jit(transport_fn, static_argnames=["forward"])
```

Comment:
i think a good test would be to transport samples, on `xs`. get `ys`, keep them in that order, and then run the Hungarian algorithm and make sure the optimal permutation with cost `h` is identity (could use scipy's https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.linear_sum_assignment.html)